lib/Github/Model/SearchIssuesResult.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace Github\Model;
+
+use function Makasim\Values\get_objects;
+use function Makasim\Values\get_value;
+
+class SearchIssuesResult
+{
+    private $values = [];
+
+    private $objects = [];
+
+    public function getTotalCount()
+    {
+        return get_value($this, 'total_count');
+    }
+
+    public function getIncompleteResults()
+    {
+        return get_value($this, 'incomplete_results');
+    }
+
+    /**
+     * @return Issue[]
+     */
+    public function getItems()
+    {
+        return iterator_to_array(get_objects($this, 'items', Issue::class));
```

Comment:
IIRC, this is the only code where you do `iterator_to_array`, can you explain why?

Reply:
`get_objects` returns generator which causes problems in ResultPager. There is array_merge function. 

I would go one step futher and make ResultPager lazy so it can use generator and there is no need for `iterator_to_array`.   I understand that this a big change and it would be better to address it in separate PR. So I've kept this PR as small and minimalistic as possible.

Reply:
Yes a separate pr for the `ResultPager` is always welcome!